sympy/stats/stochastic_process_types.py

```diff
@@ -971,37 +950,267 @@ def probability(self, condition, given_condition=None, evaluate=True, **kwargs):
+        contain a RandomIndexedSymbol of the process.
+    condition: Relational, Logic
+        The given conditions under which computations should be done.
+
+    Returns
+    =======
+
+    Expectation of the RandomIndexedSymbol.
+
+    """
+    new_expr, new_condition = _rvindexed_subs(process, expr, condition)
+
+    if not is_random(new_expr):
+        return new_expr
+    new_pspace = pspace(new_expr)
+    if new_condition is not None:
+        new_expr = given(new_expr, new_condition)
+    if new_expr.is_Add:  # As E is Linear
+        return Add(*[new_pspace.compute_expectation(
+                    expr=arg, evaluate=evaluate, **kwargs)
+                    for arg in new_expr.args])
+    return new_pspace.compute_expectation(
+            new_expr, evaluate=evaluate, **kwargs)
+
+def _probability_stoch(process, condition, given_condition=None, evaluate=True, **kwargs):
+    r"""
+    Computes probability.
+
+    Parameters
+    ==========
+
+    process : str
+       Key value of dict_rv_subs RandomVariable used to substitute
+    condition: Relational
+            Condition for which probability has to be computed. Must
+            contain a RandomIndexedSymbol of the process.
+    given_condition: Relational/And
+            The given conditions under which computations should be done.
+
+    Returns
+    =======
+
+    Probability of the condition.
+
+    """
+    new_condition, new_givencondition = _rvindexed_subs(process, condition, given_condition)
+
+    if isinstance(new_givencondition, RandomSymbol):
+        condrv = random_symbols(new_condition)
+        if len(condrv) == 1 and condrv[0] == new_givencondition:
+            return BernoulliDistribution(_probability_stoch(process, new_condition), 0, 1)
+
+        if any([dependent(rv, new_givencondition) for rv in condrv]):
+            return Probability(new_condition, new_givencondition)
+        else:
+            return _probability_stoch(process, new_condition)
+
+    if new_givencondition is not None and \
+            not isinstance(new_givencondition, (Relational, Boolean)):
+        raise ValueError("%s is not a relational or combination of relationals"
+                % (new_givencondition))
+    if new_givencondition == False:
+        return S.Zero
+    if new_condition == True:
+        return S.One
+    if new_condition == False:
+        return S.Zero
+    if not isinstance(new_condition, (Relational, Boolean)):
+        raise ValueError("%s is not a relational or combination of relationals"
+                % (new_condition))
+    if new_givencondition is not None:  # If there is a condition
+    # Recompute on new conditional expr
+        return _probability_stoch(process, given(new_condition, new_givencondition, **kwargs), **kwargs)
+    result = pspace(new_condition).probability(new_condition, **kwargs)
+    if evaluate and hasattr(result, 'doit'):
+        return result.doit()
+    else:
+        return result
+
+def given_parser(expr, condition):
```

Comment:
I don't think this should be in global namespace. Can you give it a better name?

Reply:
I agree this function should not be in the global namespace, also, can this be named as `_get_const_timerv`?. Also, if this plan looks good then should I add docstrings?

Reply:
Yes you can add docstrings.